pkg/appclient/jsonrpc.go

```diff
@@ -0,0 +1,116 @@
+package appclient
+
+import (
+	"bytes"
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/cometbft/cometbft/crypto"
+
+	"pocket/x/service/types"
+	sessiontypes "pocket/x/session/types"
+	sharedtypes "pocket/x/shared/types"
+)
+
+// handleJSONRPCRelays handles JSON RPC relay requests.
+func (app *appClient) handleJSONRPCRelays(
+	ctx context.Context,
+	serviceId string,
+	request *http.Request,
+) (responseBz []byte, err error) {
+	// Read the request body bytes.
+	payloadBz, err := io.ReadAll(request.Body)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hash and sign the request payload.
+	hash := crypto.Sha256(payloadBz)
+	signature, _, err := app.keyring.Sign(app.keyName, hash)
```

Comment:
MAJOR BLOCKER!

This feels incorrect.

We need to:
1. Create the `RelayRequest` protobuf
2. Sign it (without the signature)
3. Insert the signature

TIPS:
- RelayRequest should have a function (where it is a receiver) that signs things
- The RelayRequest object itself will be responsible for removing the fields that don't need to be part of the signature.

This is how we can enforce consistent signature addition & verification.

Otherwise, it feels like we're signing an unstructured hash, which is hard to track & verify.

Reply:
I think as I didn't get to work on this last week when I bring this into main I will just be adding rings in straight away instead of a separate PR. This is true tho. We need the method `GetSignableBytes()` on the request and response and to do it in this fashion 👍🏼